six/modules/c++/six/include/six/ByteProvider.h

```diff
@@ -50,6 +51,25 @@ namespace six
 class ByteProvider : public nitf::ByteProvider
 {
 public:
+
+    /*!
```

Comment:
When is it valid to default construct a ByteProvider?  If it's only by an inheriting class, this should be protected.

Reply:
Agreed. The default constructor is now protected.